File: app/src/main/java/org/ar25/androidno/presenters/DaggerPresenterModule.java
package org.ar25.androidno.presenters;

import android.app.Application;

import org.ar25.androidno.NOApplication;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;


@Module
public class DaggerPresenterModule {

  NOApplication mApplication;

  public DaggerPresenterModule(Application application) {
    mApplication = (NOApplication) application;
  }

  @Provides @Singleton
  MainPresenter provideMainPresenter() {
    MainPresenter presenter = new MainPresenter();

    mApplication.getNOAppComponent().inject(presenter);

    return presenter;
  }
}

File: app/src/main/java/org/ar25/androidno/presenters/DaggerPresenterModule.kt
package org.ar25.androidno.presenters

import org.ar25.androidno.NOApplication

import javax.inject.Singleton

import dagger.Module
import dagger.Provides


@Module
class DaggerPresenterModule {

    @Provides @Singleton
    internal fun provideMainPresenter(): MainPresenter {
        val presenter = MainPresenter()

        NOApplication.noAppComponent.inject(presenter)

        return presenter
    }

    @Provides @Singleton
    internal fun provideDetailPresenter(): DetailPresenter {
        val presenter = DetailPresenter()

        NOApplication.noAppComponent.inject(presenter)

        return presenter
    }
}
